Library/Homebrew/cmd/shellenv.sh

```diff
@@ -11,9 +11,10 @@ homebrew-shellenv() {
       echo "set -gx HOMEBREW_PREFIX \"$HOMEBREW_PREFIX\";"
       echo "set -gx HOMEBREW_CELLAR \"$HOMEBREW_CELLAR\";"
       echo "set -gx HOMEBREW_REPOSITORY \"$HOMEBREW_REPOSITORY\";"
-      echo "set -g fish_user_paths \"$HOMEBREW_PREFIX/bin\" \"$HOMEBREW_PREFIX/sbin\" \$fish_user_paths;"
-      echo "set -q MANPATH; or set MANPATH ''; set -gx MANPATH \"$HOMEBREW_PREFIX/share/man\" \$MANPATH;"
-      echo "set -q INFOPATH; or set INFOPATH ''; set -gx INFOPATH \"$HOMEBREW_PREFIX/share/info\" \$INFOPATH;"
+      echo "contains \"$HOMEBREW_PREFIX/sbin\" \$PATH; or set -gx PATH \"$HOMEBREW_PREFIX/sbin\" \$PATH;"
```

Comment:
Can we kill the `contains`? We don't do that for anything else and it seems overkill.

Reply:
It is not overkill, it should be done for the rest as well.

Fish loads its config *always*, whether login or not, interactive or not. That means this could easily result in growing $PATH.

Reply:
Hi @MikeMcQuaid, I didn't want to modify the shellenv output too much and was planning to not use `contains`, but discussion on the [fish gitter](https://gitter.im/fish-shell/fish-shell) led me to include it. As @faho explained the PATH variables will continue to grow if you open fish within fish, for example. Thanks for the review.

Reply:
> As @faho explained the PATH variables will continue to grow if you open fish within fish, for example.

This also applies to Bash and ZSH and no-one has complained about this until this PR.

Can we scope this just to `PATH` for now, please.

Reply:
Done. Scope is limited to `PATH` variable for fish shell only.